backends/intel_gpu/kernels/kernels.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "paddle/phi/capi/all.h"
+
+namespace custom_kernel {
+template <typename T>
+void TransposeKernel(const phi::Context& ctx,
+                     const phi::DenseTensor& x,
+                     const std::vector<int>& axis,
+                     phi::DenseTensor* out);
+
+template <typename T>
+void TransposeKernelGPU(const phi::Context& ctx,
+                     const phi::DenseTensor& x,
+                     const std::vector<int>& axis,
+                     phi::DenseTensor* out);
+
+template <typename T>
+void SoftmaxKernel(const phi::Context& dev_ctx,
+                   const phi::DenseTensor& x,
+                   int axis,
+                   phi::DenseTensor* out);
+}  // namespace custom_kernel
```

Comment:
This header file of "kernels.h" is needless?

Reply:
forward declaration required , se we need this file